js-advancers/gutenberg
# Add/controls template esnext

## Description

Add in new template for Create Block package, using [Gutenberg Examples - 04 Controls ESNext](https://github.com/WordPress/gutenberg-examples/tree/master/04-controls-esnext)
Example block written with ESNext standard and JSX support for GUI Controls– build step required.

## How has this been tested?




1. `cd gutenberg`
2. `npx wp-create-block -t controls-esnext`
3. Quick option: default for all settings; Personalized option: Client's choice
4. run build _(optional? not needed.)_
5. Finder -> open Gutenberg folder directory (leave folder open to clone/move, Step 7)
6. Deploy a local/_remote_ WordPress site
7. Clone/_move_ generated _**example-04-controls-esnext**_ plugin folder (from Step 5) into your local/_remote_  wp-content/plugins folder
8. WordPress Dashboard -> plugins -> activate _Example: Controls (ESNext)_
9. Block will appear as _Example: Controls (ESNext)_ in editor under _Layout_; add the block
10. Verify in browser

## Screenshots 
Step 8
<img width="979" alt="Screen Shot 2020-06-19 at 12 24 40 PM" src="https://user-images.githubusercontent.com/984108/85173588-6d469680-b228-11ea-8da4-e12487062180.png">
Step 9
<img width="164" alt="Screen Shot 2020-06-19 at 12 25 56 PM" src="https://user-images.githubusercontent.com/984108/85173611-7b94b280-b228-11ea-87ba-8517d6ddb645.png">
![Screen Shot 2020-06-20 at 1 32 02 AM](https://user-images.githubusercontent.com/984108/85197487-15df0f80-b296-11ea-8e86-47b481b1c8a3.png)
Step 10
![Screen Shot 2020-06-20 at 1 32 15 AM](https://user-images.githubusercontent.com/984108/85197496-21cad180-b296-11ea-9e6e-954f3df7f8af.png)

## Types of changes

Add - new example template




## Checklist:

## Commit Log:
- add/controls-template-esnext - initial commit; clone exnext default sample
- add/controls-template-esnext - update sample from wp/gutenberg master
- add/controls-template-esnext - update to match https://github.com/WordPress/gutenberg-examples/tree/master/04-controls-esnext sample
- add/controls-template-esnext - reverted edits from commit 9010591; removed unneeded import js; add back in the scss to the template block folder; prevent missing css error; all of this will likely change with gutenberg and create block update; adjusted background colors for editor view and client view to easily distinguish block is active
- add/controls-template-esnext - reverted mistakened edit of esnext template
